controller/lib/core/blower_fsm.h

```diff
@@ -155,11 +155,109 @@ class PressureControlFsm {
   Time expire_end_;
 };
 
+// "Breath finite state machine" for pressure assist mode.
+//
+// This mode is used for a patient that is not fully sedated and can initiate
+// breaths by himself, but once a breath has started, it is fully controlled
+// by the ventilator: inspiration time is set by the ventilator.
+// This mode enforce a minimum respiratory rate in case the patient does not
+// initiate enough breaths.
+// Similarly to pressure control, uses a square wave, only the length of the
+// exhale leg can vary depending on the patient breath initiation.
+// Unlike pressure control, this fsm starts with exhale and ends at the end of
+// inhale to be able to start the first inhale on patient trigger.
```

Comment:
Why not start the first inhale right away as a mandatory breath, as if the patient had been sedated? That would make this mode more uniform with pressure control and we could always assume that a new breath starts with an inhale. And we wouldn't have unpleasant effects when changing modes - double inhale or exhale. I also keep stumbling when reading the code because of this, e.g. because inspire_end_ > latest_expire_end_.

I think if you change it, desired_state() is very simple - just follow the square wave. The main logic of the mode then is in finished() which becomes (inspiratory effort detected) || (this breath is taking too long).